src/modules/include/moduleManager.hpp

```diff
@@ -12,7 +12,7 @@
 class ModuleManager {
 public:
     ModuleManager(const std::function<int(Message)>& pushMessage,
-                configuration::ConfigurationParser configurationParser,
+                std::shared_ptr<configuration::ConfigurationParser> configurationParser,
```

Comment:
`ConfigurationParser` could go first (or last) so the functions are grouped together. This is not wrong anyway, just a comment... if you happen to agree and make the change, the data member order could also match the argument list